jzy3d-jGL/jzy3d-jGL-awt/src/main/java/jgl/wt/awt/listener/GLUTListener.java

```diff
@@ -0,0 +1,124 @@
+   *
+   * @param target Rendering target component
+   * @param key    Generated ASCII character (i.e. 'a' or 'A' depending on SHIFT key)
+   * @param x      Window relative coordinate x of mouse when event has occurred
+   * @param y      Window relative coordinate y of mouse when event has occurred
+   */
+  default void onKeyboard(Component target, char key, int x, int y) {
+  }
+
+  /**
+   * Sets the keyboard key up callback for the current window.
+   * void glutKeyboardUpFunc (void (*func)(unsigned char key, int x, int y))
+   *
+   * @param target Rendering target component
+   * @param key    Generated ASCII character (i.e. 'a' or 'A' depending on SHIFT key)
+   * @param x      Window relative coordinate x of mouse when event has occurred
+   * @param y      Window relative coordinate y of mouse when event has occurred
+   */
+  default void onKeyboardUp(Component target, char key, int x, int y) {
+  }
+
+  /**
+   * Sets the special keyboard key callback for the current window.
+   * void glutSpecialFunc(void (*func)(int key, int x, int y))
+   *
+   * @param target Rendering target component
+   * @param key    Special key code as char - i.e. GLUT_KEY_PAGE_UP
+   * @param x      Window relative coordinate x of mouse when event has occurred
+   * @param y      Window relative coordinate y of mouse when event has occurred
+   */
+  default void onSpecialKey(Component target, char key, int x, int y) {
+  }
+
+  /**
+   * Sets the special keyboard key up callback for the current window.
+   * void glutSpecialUpFunc (void (*func)(unsigned char key, int x, int y))
+   *
+   * @param target Rendering target component
+   * @param key    Special key code as char - i.e. GLUT_KEY_PAGE_UP
+   * @param x      Window relative coordinate x of mouse when event has occurred
+   * @param y      Window relative coordinate y of mouse when event has occurred
+   */
+  default void onSpecialKeyUp(Component target, char key, int x, int y) {
+  }
+
+  /**
+   * Sets the display callback for the current window (occurs when window needs to be redisplayed).
+   * void glutDisplayFunc (void (*func)(void))
+   *
+   * @param target Rendering target component
+   */
+  default void onDisplay(Component target) {
+  }
+
+  /**
+   * Sets the reshape callback for the current window (triggered when window dimensions are changed)
+   * void glutReshapeFunc (void (*func)(int width, int height))
+   *
+   * @param target Rendering target component
+   * @param width  New width of window, px
+   * @param height New height of window, px
+   */
+  default void onReshape(Component target, int width, int height) {
+  }
+
+  /**
+   * Sets the global idle callback so that program can perform background processing tasks.
+   * void glutIdleFunc (void (*func)(void))
+   *
+   * @param target Rendering target component
+   */
+  default void onIdle(Component target) {
+  }
+
+  /**
+   * Indicates where the listener has usable {@link this#onMouse(Component, int, int, int, int)}
```

Comment:
Typo on **wether**?

I am surprised by this method. Is this something that appeared in GLUT, or are you following a pattern?

Reply:
> I am surprised by this method. Is this something that appeared in GLUT, or are you following a pattern?

There are some GLUT initialisations happening if listener supports certain capability